hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/cache/TableCacheImpl.java

```diff
@@ -70,7 +77,10 @@ public void put(CACHEKEY cacheKey, CACHEVALUE value) {
 
   @Override
   public void cleanup(long epoch) {
-    executorService.submit(() -> evictCache(epoch));
+    // If it is never do nothing.
```

Comment:
The Cache doesn't know it is being invoked on flush.. so this check is misleading.

Reply:
It is caller responsibility to call this method after the flush. And right now this is called only after flushing to DB from OzoneManagerDoubleBuffer.

So, if it is NEVER, we will not do the cleanup.